subprojects/docs/src/snippets/files/fileCollections/kotlin/build.gradle.kts

```diff
@@ -53,25 +55,27 @@ tasks.register("usage") {
         val file: File = collection.singleFile
 
         // Add and subtract collections
-        val union = collection + layout.files("src/file2.txt")
-        val difference = collection - layout.files("src/file2.txt")
+        val union = collection + projectLayout.files("src/file2.txt")
+        val difference = collection - projectLayout.files("src/file2.txt")
 
         // end::usage[]
     }
 }
 
 tasks.register("filterTextFiles") {
+    val collection: FileCollection = collection
```

Comment:
Global `collection` val is not serializable because of this string (["src/file1.txt"](https://github.com/gradle/gradle/pull/21840/files#diff-d62861ebaef012097ced1b2fe50d5096d998722e5d3b62e274b29d97fcce6252R5)). 
After reading the documentation, I found that I should do something like this ("A task input or output property or a script variable to capture the result...").
However I found that this way of doing things wasn't required in the [groovy script](https://github.com/gradle/gradle/pull/21840/files#diff-a383c8e4dcc100282cb625e8f00baed1d01c9168f08ccecffd2bc1a5e434a02fR64) to make it work. So, I'm not sure if this an expected behaviour or I'm doing something wrong? 🤔

Reply:
"Global" val in Kotlin is compiled to a "property on a build script", so when lambda captures that it captures buildscript object itself and calls capturedBuildScript.getCollection.
For instance, Kotlin has no guarantee that every `.collection` call would produce the same result

When you do `val collection = collection` it creates local variable, and suddenly Kotlin knows it can't change, so lambda can capture the value itself.

Reply:
WDYT of adding a comment like `// copy collection field to a local variable for configuration cache support`?

Reply:
Thank you for the detailed explanation about what's going on with the `collection` property. 
I think is good to have that comment in the snippet, so I'll do it tomorrow as soon as I can 👍 .